### DISCUSSION TITLE
accesso remoto senza ip pubblico

### DISCUSSION THREAD

**Discussion by Matteogolf** [indomus/forum#1490] [HUB: Home Assistant]
Ciao, ho installato su homeassistant duckdns e nginx.
Ho creato il mio dominio su duckdns, ho accettato la licenza ed inserito l'url e token in duckdns poi in nginx il dominio.
Ho aperto nel modem la porta esterna 443 ed interna 443 sull'ip di HA.
Ho linea internet fastweb senza IP pubblico.

Ho poi seguito la guida "Aggiornare il proprio record su DuckDNS tramite automazione Home Assistant" pensando di ovviare al problemi dell'ip dinamico.

Purtroppo digitando il mio dominio.duckdns.org non riesco a collegarmi da remoto.

Potete per favore darmi una mano?

grazie,

ciao

**Comment by KC-inDomus** [comment]
se non hai ip pubblico devi usare questa [guida](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-che-usino-nat/).

**Reply by Matteogolf** [reply]
grazie , provo subito.

ciao

**Reply by Matteogolf** [reply]
ho creato il server virtuale.
Ho messo in comunicazione il server con duckdns ma credo ci sia un problema.
Nel test:
./duck.sh
cat duck.log
anzichè comparire "OK" mi compare "KO" ed il mio nome scelto.
ho seguito la guida passo passo ed ho inserito nella stringa il mio nome duckdns ed il token.

**Reply by snakuzzo** [reply]
fai vedere cosa hai scritto

**Reply by Matteogolf** [reply]
mkdir duckdns
cd duckdns
touch duck.sh
chmod +x duck.sh
nano duck.sh
echo url="https://www.duckdns.org/update?domains=<mio_nome_dominio>&token=<mio_token>&ip=" | curl -k -o ~/duckdns/duck.log -K -
export VISUAL=nano; crontab -e
*/5 * * * * ~/duckdns/duck.sh >/dev/null 2>&1 (in fondo all'editor)
sudo service cron start

**Reply by Matteogolf** [reply]
nella stringa "echo url" ho inserito il mio nome dominio senza .duckdns.org ed il token

**Reply by snakuzzo** [reply]
Togli `&ip=` e riprova

**Reply by Matteogolf** [reply]
ho provato ma purtroppo l'esito è ancora KO.
grazie

**Reply by snakuzzo** [reply]
Scusa, ma se da riga di comando dai...

```
curl https://www.duckdns.org/update?domains=<mio_nome_dominio>&token=<mio_token>
```

...che ti dà?

**Reply by Matteogolf** [reply]
-bash: syntax error near unexpected token `newline'

**Reply by snakuzzo** [reply]
Hai sbagliato a scrivere il comando...ritenta

**Reply by snakuzzo** [reply]
Prova così...mettendo la URL tra apici...

```
curl 'https://www.duckdns.org/update?domains=<mio_nome_dominio>&token=<mio_token>'
```

**Reply by Matteogolf** [reply]
mi scrive:
KOmio_nome

**Reply by snakuzzo** [reply]
Mi fai vedere che cosa stai scrivendo per favore? Fai uno screenshot oscurando parte del token e del dominio

**Reply by snakuzzo** [reply]
Mica stai mettendo pure i simboli maggiore e minore?

**Reply by Matteogolf** [reply]
ok, ora mi vergognerò... si mettevo maggiore e minore.

Ora funzione!

grazie mille

**Comment by Matteogolf** [comment]
Sono al punto di dover installare il client openVPN nel raspberry solo che, da terminale, quando digito apt-get mi dice che non conosce il comando.
è capitato anche ad altri?
Cercando su google ho trovato quanlche metodo ma non ha funzionato.
Suggerimenti?

grazie,
ciao

**Reply by snakuzzo** [reply]
Se sei su RaspOS o comunque distro debian like è impossibile

**Reply by Matteogolf** [reply]
dentro home assistant dal terminale SSH digito:
sudo apt-get install openvpn
e ricevo:
sudo: apt-get: command not found

**Reply by snakuzzo** [reply]
Forse ti sei perso un pezzo.
L'installazione di un client openvpn su HassOS non si fa così.
quella guida che stai vedendo è per RaspOS! :)

Per HassOS devi fare in modo di trovare un addon tipo... https://github.com/larsklitzke/homeassistant-openvpn-client
...ma devi provare perché è di circa 3 anni fa.

**Reply by Matteogolf** [reply]
noooo, solo 6 ore per configurare il server&co :-)

provo col link suggerito, ho avviato da ssh:
cd /addons && git clone https://github.com/larsklitzke/homeassistant-openvpn-client.git

creato pannello openvpn:
panel_iframe:
  openvpn:
    title: OpenVPN
    icon: mdi:cloud-check
    url: http://IPhomeassistant:8090 

riavviato homeassistant

se clicco il pulsante OpenVpn compare:
Connessione negata da iphomeassistant

**Reply by snakuzzo** [reply]
Ripeto...è roba molto vecchia...noi possiamo fare ben poco.
Prova a guardare le issue del progetto e nel caso aprine una tu.

**Reply by Matteogolf** [reply]
ok, grazie comunque per tutti gli aiuti dati! ;-)

**Reply by vinny74man** [reply]
Scusa ma IP pubblico di fastweb non costa nulla devi solo chiederlo, io lo tengo da quando ho fatto l'abbonamento con loro, quindi non vedo il problema chiama e fallo abilitare di solito entro 24 ore è attivo

**Reply by Matteogolf** [reply]
> Scusa ma IP pubblico di fastweb non costa nulla devi solo chiederlo, io lo tengo da quando ho fatto l'abbonamento con loro, quindi non vedo il problema chiama e fallo abilitare di solito entro 24 ore è attivo

Ho letto che l’io pubblico è meno sicuro e per quello cercare un metodo alternativo.

**Reply by snakuzzo** [reply]
Informazioni sommarie e fuorivianti. Ma dove le hai lette ste cavolate?
Se vuoi stare sicuro, stacca internet :D ... a quel punto sei a posto.
La sicurezza non è legata esclusiavamente al fatto di avere ip pubblico o meno.
Ascolta il consiglio che già ti è stato dato...fatti dare l'ip pubblico e campa sereno ;)
